File: app/src/main/java/com/billyji/datenight/data/FoodSelectionDataModel.java
package com.billyji.datenight.data;

public class FoodSelectionDataModel
{
    private static int m_maxDistance;
    private static double m_minStars;
    private static String m_maxPrice;

    private static final int MAX_DISTANCE_ALLOWED = 20;
    private static final double MAX_STARS_ALLOWED = 5;
    private static final int MAX_PRICE_ALLOWED = 5;
    private static final String MAX_PRICE = "4.5";

    public static void setMaxPrice(String price)
    {
        m_maxPrice = Integer.parseInt(price) >= MAX_PRICE_ALLOWED ? MAX_PRICE : price;
    }

    public static void setMaxDistance(int distance)
    {
        m_maxDistance = distance > MAX_DISTANCE_ALLOWED ? MAX_DISTANCE_ALLOWED : distance;
    }

    public static void setMinStars(double stars)
    {
        m_minStars = stars >= MAX_STARS_ALLOWED ? 4.5 : stars;
    }

    public static int getMaxDistance()
    {
        return m_maxDistance;
    }

    static String getMaxPrice()
    {
        return m_maxPrice;
    }

    static double getMinStars()
    {
        return m_minStars;
    }
}

File: app/src/main/java/com/billyji/datenight/data/FoodSelectionDataModel.kt
package com.billyji.datenight.data

object FoodSelectionDataModel {
    private var distanceMap: HashMap<Int, Int> = hashMapOf(0 to 1, 1 to 2, 2 to 5, 3 to 10, 4 to 20)

    internal var maxDistance: Int = 0
        set(distance) {
            field = distanceMap[distance]!!
        }
    internal var minStars: Double = 0.toDouble()
    internal var maxDollarSigns: Int = 0
}
